packages/types/src/auth/HttpSigner.ts

```diff
@@ -1,6 +1,20 @@
-import { HttpRequest } from "../http";
+import { HttpRequest, HttpResponse } from "../http";
 import { Identity } from "../identity/identity";
 
+/**
+ * @internal
+ */
+export interface ErrorHandler {
+  (signingProperties: Record<string, unknown>): <E extends Error>(error: E) => never;
+}
+
+/**
+ * @internal
+ */
+export interface SuccessHandler {
+  (httpResponse: HttpResponse | unknown, signingProperties: Record<string, unknown>): void;
```

Comment:
signingProperties should have a type. It can be a record too (`&`), but it should also indicate that it tends to have properties like `name`, `in`, etc.

Reply:
This is the generic `HttpSigner` which has no context for the properties of any specific `HttpAuthScheme`.

I think concrete implementations of `HttpSigner` can provide more type hints, but I can add that in a follow-up PR.